examples/security/oauth/exposer.js

```diff
@@ -0,0 +1,53 @@
+/********************************************************************************
+ * Copyright (c) 2018 - 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the W3C Software Notice and
+ * Document License (2015-05-13) which is available at
+ * https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR W3C-20150513
+ ********************************************************************************/
+
+let td = {
+    "@context": "https://www.w3.org/2019/wot/td/v1",
+    "title": "OAuth",
+    "id": "urn:dev:wot:oauth:test",
+    "securityDefinitions": {
+        "oauth2_sc": {
+            "scheme": "oauth2",
+            "flow": "client_credentials",
+            "authorization": "https://example.com/authorization",
+            "token": "https://localhost:3000/token",
+            "scopes": [
+                "limited",
+                "special"
+            ]
+        }
+    },
+    "security": [
+        "oauth2_sc"
+    ],
+    "actions": {
+        "sayOk": {
+            "forms": [
+                {
+                    "href": "http://localhost:3000/resource",
+                    "htv:methodName": "GET"
```

Comment:
Is this supported in node-wot? So does the HTTP binding start listening to GET for actions?

Reply:
See this comment above:
> the example provided requires that node-wot uses the form provided in the produce dictionary. So in the future, I might review it to make it work at 100%.

Since we don't have yet a server-side implementation of oAuth the point of the example was to use the mockup oAuth server as `invokeaction` operation. So right now the example does not validate the token. I guess when we can override href inside forms using the *produce dictionary* the problem should be solved. What do you think?

Reply:
I see, sorry for missing the comment. So does this exposed Thing contain a form that doesn't belong to itself? I am pretty sure that node-wot would replace with its own.

Reply:
Exactly, I imagined that in the future TDs could contain forms that are not served from the exposing servient. Is that so? otherwise, I'll be happy to change the example.